shaarli_client/utils.py

```diff
@@ -21,3 +22,27 @@ def generate_all_endpoints_parsers(subparsers, endpoints):
     """Generate all endpoints' subparsers from an endpoints dict"""
     for ep_name, ep_metadata in endpoints.items():
         generate_endpoint_parser(subparsers, ep_name, ep_metadata)
+
+
+def write_output(args, response):
+    """Write the API response to file or standard output"""
+    if not args.outfile:
+        if args.format == 'json':
+            print(response.json())
+        elif args.format == 'pprint':
+            print(json.dumps(response.json(), sort_keys=True, indent=4))
+        elif args.format == 'text':
+            print(response.text)
+    else:
+        try:
```

Comment:
test scenarii ideas:
- check a file is written
- for each supported format:
    - check a file is written
    - check an existing file is overwritten
    - check the file can be read, and its output loaded using the corresponding Python module